lib/client/checks/http/http-executor.ts

```diff
@@ -0,0 +1,90 @@
+import * as logger from '../../../log';
+import * as version from '../../../version';
+import { axiosInstance } from '../../../axios';
+import { retry } from '../../retry/exponential-backoff';
+import type { AxiosResponse } from 'axios';
+import type { CheckId, CheckResult, CheckStatus } from '../types';
+
+export async function executeHttpRequest(
+  checkOptions: {
+    id: CheckId;
+    name: string;
+  },
+  httpOptions: {
```

Comment:
should probably allocate for headers values to be passed there too. I think we'll soon have more elaborate checks that are likely to require auth headers and the likes

Reply:
good idea, I'll incorporate it in the next iteration.